File: src/main/java/io/kragelv/library/service/AuthorService.java
package io.kragelv.library.service;

import java.util.UUID;

import io.kragelv.library.dto.author.AuthorDto;
import io.kragelv.library.dto.author.AuthorRequestDto;
import io.kragelv.library.dto.page.PageResponse;

public interface AuthorService {

    PageResponse<AuthorDto> getList(int page, int limit);

    AuthorDto getById(UUID id);

    UUID create(AuthorRequestDto dto);

    AuthorDto update(UUID id, AuthorRequestDto dto);

    void deleteById(UUID id); 
}

File: client/src/services/AuthorService.ts
import { AxiosResponse } from 'axios';

import { AuthorDto } from '../dtos/author/AuthorDto';
import { AuthorRequestDto } from '../dtos/author/AuthorRequestDto';
import { PageQueryParams } from '../dtos/page/PageQueryParams';
import { PagedResponse } from '../dtos/page/PageResponse';
import $api from '../http';

class AuthorService {
  static readonly DEFAULT_LIMIT = 15;

  static async getPage(params: PageQueryParams): Promise<AxiosResponse<PagedResponse<AuthorDto>>> {
    const limit = params.limit === this.DEFAULT_LIMIT ? undefined : params.limit;
    return $api.get('/v1/authors', {
      params: { page: params.page, limit },
    });
  }

  static async getById(id: string): Promise<AxiosResponse<AuthorDto>> {
    return $api.get(`/v1/authors/${id}`);
  }

  static async create(createAuthorDto: AuthorRequestDto): Promise<AxiosResponse<string>> {
    return $api.post('/v1/authors', createAuthorDto);
  }

  static async update(
    id: string,
    updateAuthorDto: AuthorRequestDto,
  ): Promise<AxiosResponse<AuthorDto>> {
    return $api.put(`/v1/authors/${id}`, updateAuthorDto);
  }

  static async delete(id: string): Promise<AxiosResponse<void>> {
    return $api.delete(`/v1/authors/${id}`);
  }
}

export default AuthorService;
